### PR TITLE
Edited SkillsPage Pageclass and added EDitSkillTest test class to add…

### PR DESCRIPTION
**Pull request by Vishnu-Kartha**
… edit a skill or description.

### CODE REVIEW

File: src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java [hunk 1]
```diff
@@ -55,4 +63,28 @@ public void verifySkill(String skillName, String skillDescription){
...
+    public void editSkill(String oldSkillName, String newSkillName, String oldSkillDescription, String newSkillDescription){
+        String checkSkillXpath = checkSkillPartialXpath + oldSkillName + "']";
+        String editbuttonXpath = checkSkillPartialXpath + oldSkillName + editButtonXPathLastPart;
+        if (isElementPresent(By.xpath(checkSkillXpath))){
+            click(By.xpath(editbuttonXpath));
+            WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
+            wait.until(ExpectedConditions.elementToBeClickable(By.xpath(nameFieldXpath)));
+            WebElement namefield = driver.findElement(By.xpath(nameFieldXpath));
+            WebElement descriptionfield = driver.findElement(By.xpath(descriptionFieldXpath));
+            namefield.sendKeys(Keys.CONTROL + "a");
```
Comment (RahulNair92) [inline src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java:75]: Any reason why common method setText was not used

File: src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java [hunk 2]
```diff
@@ -55,4 +63,28 @@ public void verifySkill(String skillName, String skillDescription){
...
     }
 
+    public void editSkill(String oldSkillName, String newSkillName, String oldSkillDescription, String newSkillDescription){
+        String checkSkillXpath = checkSkillPartialXpath + oldSkillName + "']";
+        String editbuttonXpath = checkSkillPartialXpath + oldSkillName + editButtonXPathLastPart;
+        if (isElementPresent(By.xpath(checkSkillXpath))){
+            click(By.xpath(editbuttonXpath));
+            WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
+            wait.until(ExpectedConditions.elementToBeClickable(By.xpath(nameFieldXpath)));
+            WebElement namefield = driver.findElement(By.xpath(nameFieldXpath));
```
Comment (RahulNair92) [inline src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java]: Why sttext method from pagebase was not used instead of sendkeys

Comment (Vishnu-Kartha) [reply src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java]: Using set text adds the input in addition to the already present input. Web element is used to find and clear the input field and set text is used after clearing the field.

File: src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java [hunk 3]
```diff
@@ -25,6 +30,9 @@ public SkillsPage(WebDriver _driver){
     public String checkSkillPartialXpath = "//div[@style='flex: 2 1 0%;']/following::div[text()='";
     public String checkdescPartialXpath = "//div[@style='flex: 4 1 0%;']/following::div[text()='";
 
+    public String editButtonXPathLastPart = "']//following::i[@class='oxd-icon bi-pencil-fill']";
+    //div[text()='Selenium']//following::i[@class='oxd-icon bi-pencil-fill']
```
Comment (RahulNair92) [inline src/test/java/ca/qaguru/oranghrmbatch27/pages/SkillsPage.java]: remove comments

**Review by RahulNair92 — APPROVED** [review]

**Pull request merged**